core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/QcCorrespondingField.java

```diff
@@ -1,5 +1,69 @@
 package uk.ac.bbsrc.tgac.miso.core.data;
 
 public enum QcCorrespondingField {
-  CONCENTRATION, VOLUME, NONE;
+  CONCENTRATION {
+
+    @Override
+    public void updateField(Pool pool, double value, String units) {
+      pool.setConcentration(value);
+      pool.setConcentrationUnits(units);
+    }
+
+    @Override
+    public void updateField(Library library, double value, String units) {
+      library.setInitialConcentration(value);
+      library.setConcentrationUnits(units);
+    }
+
+    @Override
+    public void updateField(Sample sample, double value, String units) {
+      sample.setConcentration(value);
+      sample.setConcentrationUnits(units);
+    }
+
+  },
+  VOLUME {
+
+    @Override
+    public void updateField(Pool pool, double value, String units) {
+      pool.setVolume(value);
+      pool.setVolumeUnits(units);
+    }
+
+    @Override
+    public void updateField(Library library, double value, String units) {
+      library.setVolume(value);
+      library.setVolumeUnits(units);
+    }
+
+    @Override
+    public void updateField(Sample sample, double value, String units) {
+      sample.setVolume(value);
+      sample.setVolumeUnits(units);
+    }
+
+  },
+  NONE;
+
+  public void updateField(Pool pool, double value, String units) {
+    throw new UnsupportedOperationException("Method not implemented for unspecified field");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Method not implemented for unspecified field" 5 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)